lib/ModelMixins/GroupMixin.ts

```diff
@@ -69,15 +55,50 @@ function GroupMixin<T extends Constructor<Model<GroupTraits>>>(Base: T) {
      * list of members in `GroupMixin#members` and `GroupMixin#memberModels`
      * should be complete, but the individual members will not necessarily be
      * loaded themselves.
+     *
+     * This returns a Result object, it will contain errors if they occur - they will not be thrown.
+     * To throw errors, use `(await loadMetadata()).throwIfError()`
+     *
+     * {@see AsyncLoader}
      */
-    async loadMembers(): Promise<void> {
+    async loadMembers(): Promise<Result<void>> {
       try {
-        if (CatalogMemberMixin.isMixedInto(this)) await this.loadMetadata();
-        await this._memberLoader.load();
-      } finally {
+        // Call loadMetadata if CatalogMemberMixin
+        if (CatalogMemberMixin.isMixedInto(this))
+          (await this.loadMetadata()).throwIfError();
```

Comment:
instead of Result.throwIfError and then catching it later to return another Result, another approach may be to have a something like a `Result.bimap(onError:(result:Result) => Result, onValue:(result:Result => Result))`. There's a lot of places where this is being used though so it might be a "something to consider later" kind of thing

Reply:
Yeah I agree, definitely worth looking at this kind of thing later. It would be nice to have an async API to chain `Results` maybe like https://www.npmjs.com/package/neverthrow#resultasyncmaperr-method